modules/fundamental/src/sbom/service/sbom.rs

```diff
@@ -933,4 +936,43 @@ mod test {
 
         Ok(())
     }
+
+    #[test_context(TrustifyContext)]
+    #[test(tokio::test)]
+    async fn versions_filtering(ctx: &TrustifyContext) -> Result<(), anyhow::Error> {
+        let _sbom1 = ctx
+            .ingest_document("quarkus-bom-2.13.8.Final-redhat-00004.json")
+            .await?;
+
+        let _sbom2 = ctx.ingest_document("ubi9-9.2-755.1697625012.json").await?;
+
+        let service = SbomService::new(ctx.db.clone());
+
+        // Test exact version match
+        let fetched = service
+            .fetch_sboms(
+                q("versions=2.13.8.Final-redhat-00004"),
+                Paginated::default(),
+                (),
+                &ctx.db,
+            )
+            .await?;
+        log::debug!("Exact version match results: {:#?}", fetched.items);
+        assert_eq!(1, fetched.total);
+
+        // Test version substring search
+        let fetched = service
+            .fetch_sboms(q("versions~2.13.8"), Paginated::default(), (), &ctx.db)
+            .await?;
+        log::debug!("Version substring search results: {:#?}", fetched.items);
+        assert_eq!(1, fetched.total);
+
+        // Test all SBOMs without version filter
+        let fetched = service
+            .fetch_sboms(Query::default(), Paginated::default(), (), &ctx.db)
+            .await?;
+        assert_eq!(2, fetched.total);
+
+        Ok(())
```

Comment:
I would encourage you to reorganize this test to look more like [this test](https://github.com/trustification/trustify/blob/60ff46581c36ec09814f1b64067ab0d92d3fce39/modules/fundamental/src/sbom/endpoints/test.rs#L664-L688). Specifically, use `ctx.ingest_documents`, create a `query` closure that can take various "q" values, and then call it for all the queries you need to support, including a full text search, e.g. `q=fbf470d`.

Reply:
@gildub You may as well just add more `query` calls to `query_sboms_by_array_values` cited above. You can ingest a 3rd doc (the `_tmp.json` one, but maybe rename it something like `etc/test-data/cyclonedx/tc-2564_tmp.json`), and all the existing query expectations should still work, I think.

Reply:
`versions` becomes just another array field to test in there with `suppliers` and `authors`